pkg/controller/middleware/auth.go

```diff
@@ -36,6 +36,21 @@ import (
 // is set in the template map. It fetches a user from the session and stores the
 // full record in the request context.
 func RequireAuth(ctx context.Context, client *auth.Client, db *database.Database, h *render.Renderer, ttl time.Duration) mux.MiddlewareFunc {
+	return requireAuth(ctx, client, db, h, ttl, true /*requireVerified*/)
+}
+
+// RequireAuthNotVerified requires a user to be logged in. Does not require the user to have verified the login email.
+func RequireAuthNotVerified(ctx context.Context, client *auth.Client, db *database.Database, h *render.Renderer, ttl time.Duration) mux.MiddlewareFunc {
+	return requireAuth(ctx, client, db, h, ttl, false /*requireVerified*/)
```

Comment:
Middleware-wise, it's better to say this requires `RequireAuth` to run first, then do something like

```golang
user := controller.UserFromContext(ctx)
if user == nil {
  // unauth, probably an error
}

// Logic as below.
client.GetUserByEmail(ctx, user.email)
```

Reply:
I could do that and have something like RequireEmailVerified, but the reason I have not is that I want checking both to be the default (so it can't be forgotten on pages as we add them).

The email verify page is special in that it requires auth, but not verified

Reply:
Fair. I'm just really worried about having a method with a magical boolean that changes the flow logic.